### ISSUE TITLE
CR-10S Pro, Micro Swiss Hot End, Thermal Runaway (temperature drops fluctuations)

### ISSUE THREAD

**Issue by quarky42** [InsanityAutomation/Marlin#37]
EDIT:  This thread has been very helpful.  There is definitely a hardware problem here and Creality has been SUPER slow about and "round about" in regards to providing help.  Long story short:  When my heated bed comes on, the DC voltage drops from 23.4 volts down to 11 volts.   When the printer is printing, 11V is not enough for the hot end and the hot end loses temperature until the heated bed shuts off. Once the heated bed shuts off, then the hot end recovers to its set point.

RE: "Before filing an issue be sure to test the 1.1 and/or 2.0 "bugfix" branches to see whether the issue is already addressed."
-- I was directed to use the Creality_DWINTest branch because of my BLTouch V3 (March build date).  So, I'm not sure that I should or could use the bugfix branch without a lot of modifications.  I am willing to try that branch but I would need my BLTouch V3 (March) and my Micro Swiss Hot End working on it.

**My Printer:**
New CR-10S Pro (Control Board "v2.4 2018-08-13" printed on it) using Creality_DWINTest Branch pulled down 5/21/19 8:23pm MST, customized and loaded to the printer with the Arduino IDE Mods:  HotEndStock enabled, ABL_BLTouch enabled (V3 / March), AllMetalHotEnd enabled (MicroSwiss HotEnd Installed, using the stock cooling fans currently.)

One of the more recent M303 Auto PID tunes was :
M301 P25.99 I2.50 D67.61 U1
(PID tuning was ran with the nozzle 6mm off the bed.  Fan was commanded "on" through the touch screen.  I couldn't detect a difference in the fan using the M106 command or using M107 to turn it off....  not sure what was going on there. I tried specifying P0 and P1....neither controlled the part cooling fan.)  
I am rerunning the auto tune again right now.  I'll edit back when I get a chance.

I measured my thermister to be 102 kOhms both at the break out board connector and at the control board connector verifying the whole path.  Both boards solder joints look okay.  I do not see any signs of a cold solder joint.

I measured 24.3V on each of the legs of the power supply.

I measured 1.4V at the extruder, x, and y stepper pots.  Also, 1.7V at the z stepper pot.  Based on what little I was able to find, I believe these are correct.

### Description

Edited this section heavily:   After the fact I see now that when my heated bed comes on the nozzle temperature drops anywhere from 15 to 30 degrees depending on a couple of other factors.   So, it looks like a hardware problem at this point, but this thread has been VERY helpful, so if it ends up being purely hardware, I hope that the thread will live on in some form because I didn't find anything else online about the nozzle temperature dropping when the print started on a CR-10S Pro.  This may help others in their troubleshooting.

### Steps to Reproduce

1. On my CR-10S Pro, Load the Creality-DWINTest firmware and custom firmware settings described above. Perform a factory reset.  Calibrate the extruder and save the newly calculated  extruder steps value. Run the PID Auto Tune and save the results to the EEPROM.
2. Slice a simple tiny model printing at 205C degrees hot end, 60mm/s, with 60C bed temperature.
3. Print and watch it fail.

I've discovered that if the heated bed is off, then the nozzle temperature is very stable like it should be.  When the heated bed is on and is heating up during its thermal cycling the nozzle temperature drops.

**Expected behavior:** 
I expect it to hold to within 1 or 2 degrees of 205 C instead of apparently turning off the heater for 30 seconds.  I have seen other Marlin firmwares on other printers "wait" near the setpoint temperature for 5 or 10 seconds before starting a print so that it can stabilize.  This seems to start printing very shortly after reaching the set point....  There should be a 15 second (maybe more) delay before printing is started to let the temperature stabilize.  (I don't think that is the cause of the problem, just an observation.)

**Actual behavior:**
I think I already described this:  The temperature starts to drop and loses 20 to even 30 degrees 30 seconds after the first plastic hits the bed.  The system halts during the print with a thermal runaway condition because the temperature is too low in most cases.  I've only had 2 prints out of ~30 reach 100% without the error coming up, but they absolutely dropped temperature multiple times during each print.

Here's a video I shot: https://www.youtube.com/watch?v=ZvkEPcxjZHU
I may or may not have flipped the printer the bird several times in the video. That is aimed squarely at the printer that  I've wasted the most part of 3 days screwing around with this thing just trying to get it to print a simple cylinder piece.

#### Additional Information

* Included a ZIP file containing your `Configuration.h` and `Configuration_adv.h` files:
[Marlin-Creality_DWINTest_ConfigurationFiles.zip](https://github.com/InsanityAutomation/Marlin/files/3221768/Marlin-Creality_DWINTest_ConfigurationFiles.zip)
[HeadPost_plainWhite.zip](https://github.com/InsanityAutomation/Marlin/files/3222236/HeadPost_plainWhite.zip)

I don't think the Configuration_adv.h was changed from original.  It has a new modified date, but I do believe that was an accidental save with no actual change being made.  I had a typo in that file and I undid it and hit save.  That's it.

I've gone back and cleaned up this original post to try and make it more clear. Hindsight being 20/20 and all.

**Comment by InsanityAutomation** [comment]
Sounds like the part cooling fan is overpowering the hotend heater. Have you tried with the part cooling fan off until quite a bit higher? Or running a PID tune with the cooling fan at 100%?

**Comment by quarky42** [comment]
Yes, I have tuned with the part cooling fan on at 50%, 75%, 100%.... makes no difference.   On the latest version of my gcode, the part cooling fan is on at 35% at the start of the print and held at 35% the whole time.   The skirt and the first couple layers are printed at 205 degrees.  The temperature only starts to fall 30 seconds after the extrusion begins.

I can put turn the cooling fan on, preheat the printer to 205, and extrude 300mm of plastic with no loss of temperature.  It is only losing temperature during an actual print job.  I cannot "make it" lose temperature outside of a print job.

**Comment by quarky42** [comment]
Also, please note:  If the thermal runaway isn't triggered, then the temperature recovers.  It goes back up to 205 degrees and sits there for 30 seconds.   Then the temperature dives again.   The cooling fan hasn't changed in that time.    As far as I can tell the heater is turning off for 30 seconds at a time. If I am printing slow enough or get "lucky", it will recover.   The print is still ruined with that huge of a temperature swing, but if it does recover, it will go back up to 205 degrees and once it reaches 205 degrees it will sit there for about 30 seconds before it starts to dive again.   Something is causing the heater to cycle in this strange way while printing.

**Comment by quarky42** [comment]
Short of having a voltage probe hooked up to the heater line during the print, I am 95% certain that the heater is turning off for 30 seconds at a time.  It is not cycling the way I would expect it to cycle.

**Comment by InsanityAutomation** [comment]
Very strange indeed... Are you able to get a serial log when this is occurring? Might be some useful data in the temp reports. One thing to possibly try is disabling autotemp on line 143 in config_adv. It may be having an issue offsetting heater power based on extrusion.

**Comment by quarky42** [comment]
I will do both tonight and report back.  If you think of anything else you'd like me to try / check, please let me know and I'll add it to the to-do list.  :)

I can capture the serial output.   I had been using putty to connect.  I finally figured out how to resize the Pronterface window.  The first few times I tried to find the "seam" to resize it, the cursor wouldn't light up.    It wasn't usable until I could stretch it out much wider, finally.

**Comment by InsanityAutomation** [comment]
Putty is awesome! Used it for years :)

**Comment by quarky42** [comment]
I loaded the B6_250K hex file with factory default PID settings.  I have the same issues, but now the temperature drops are showing up on preheat as well (though I am doing something a bit different now...I got OctoPrint set back up so I've been using that to control the printer. I will need to    So I have graphs, logs, see what the behavior is using the touch screen.)

This latest print was done with the fan completely off.  Didn't help.

I haven't had a chance to try loading the firmware from Ardiuno IDE after turning off autotemp in configuration_adv.h yet, but I will try and get to that tomorrow.  I wanted to post the log, screenshots, and a timelapse.

Thermal drop (about 20 degrees) during print.  This screenshot is after loading BLTouch B6 250k Hex and loading the factory default M301 PIDs:
![thermal drop during print B6 250k hex default M301](https://user-images.githubusercontent.com/48300108/58533265-404ebe00-81a5-11e9-9611-3623fa2c67c5.jpg)
Same thing as above just after the print was over I turned the hot end back on and just let it sit.  This is showing that same dipping behavior.  I will need to try this from the touch screen instead and see what I get.  Before, the touchscreen commanded preheat temperatures was pretty stable. 
![thermal drop after print 205 degrees B6 250k hex default M301](https://user-images.githubusercontent.com/48300108/58533270-46449f00-81a5-11e9-944a-08a335e7b0c6.jpg)
Serial Connection Log File from OctoPrint:
[serial.log](https://github.com/InsanityAutomation/Marlin/files/3230964/serial.log)
Time Lapse Video:
[headpo_1_20190528232154.zip](https://github.com/InsanityAutomation/Marlin/files/3230966/headpo_1_20190528232154.zip)

**Comment by quarky42** [comment]
Should the heated bed be receiving 12 volts or 24 volts?   I took a voltage reading earlier on the DC relay for the heated bed and was reading ~12.3v out of the relay.  Is that right?   

As far as this next graph goes, I think I found my culprit:  The heated bed.  I turned off the heated bed and just the hot end.  The hot end is sitting stable at 205 degrees.   

So my theory at this point is that every time the heated bed is coming on my hot end dips.  Otherwise I'm at a loss for why turning off the heated bed caused the hot end to behave itself.

![Heated Bed On Left Side of Graph vs Heated Bed Off the Last 5 Minutes Right Side](https://user-images.githubusercontent.com/48300108/58535091-3bd8d400-81aa-11e9-8db5-37febf7e9131.JPG)


It held temperature steady for 10 minutes with the heated bed turned off.   I'm assuming at this point that this isn't firmware related, but since you have a lot of experience I am hoping you might steer me in the right direction?

**Comment by InsanityAutomation** [comment]
Where were youre leads when you measured 12v? Only one power supply in the system and its 24v... Power supply may have an issue and is failing under load, but only the hotend shows it as the bed is much slower to respond and change.

**Comment by quarky42** [comment]
The load side.  It only shows voltage on those connectors when the heated bed is on.  Doing some more reading I think the heated bed is 12V based?

The other side of the relay showed 24.3v constant. (I did not measure that side under load... Sounds like I need to measure the hot end temperature when the heated bed is on, but based on the behavior I am pretty sure the hot end is losing power while the bed is heating.  I am uploading a video that shows the nozzle reach temperature with the bed.  When the nozzle starts to drop the bed drops 1 degree and the nozzle only recovers after the bed has recovered.

**Comment by InsanityAutomation** [comment]
Id try to get a replacement power supply. Keep in mind load side of the SSR is +/+ not +/- think of it like a light switch on that side.

**Comment by juliandroid** [comment]
It would be interesting if the PSU is throttling (overheating). Inside you have one small fan and if it does not work (if you reverse the polarity it won't run at all)?

**Comment by quarky42** [comment]
@juliandroid  Thank you for taking the time to comment. I don't think I got a chance to add this, but yes I already checked the fan last night and it is indeed running.  The fan was pointing into the power supply instead of pointing out of the power supply.  I have it pointing out as an exhaust fan and it has made no difference in the problem at all.

I felt the cover of the power supply after the bed and hot end had been on for a while and it was warm, but not uncomfortable at all.  (So somewhere between 100 degrees and 110 degrees without getting out my IR thermometer.    I had already made sure that all power connections on the power supply and on the control board / dc relay were tight.  They were.  I didn't find any loose connections when I was initially going through the printer to do the quality control that Creality is known to not always do well.

If there is something I should check inside the power supply, I am willing to carefully do so.  At this point I think @InsanityAutomation is spot on... it is likely a weak / bad power supply.

**Comment by quarky42** [comment]
What's weird is that I can heat up both the heated bed and the nozzle at the same time.  It takes about ~7 minute to heat up if I heat up both at the same time.  The nozzle heats up a lot faster with the heated bed pre-heated or not running at all.    I do not see the same major drops when the heated bed and the nozzle are going at the same time.

@InsanityAutomation Is the bed heater "all or nothing" or is it varied by PWM in any way?    I haven't run any PID calibration on the heated bed itself.   Is there any value in that?  I'm just using the defaults.   I did load the 250k hex version of the firmware last night and had no change in the behavior.  Since others had confirmed that they used the hex version without problems I figured I would try a "known good" firmware that way to make sure it wasn't something weird in the Arduino IDE project.   I did redownload the github version and I made the change that you suggested a few posts back. I did not have time to load it up and try it.  I am still willing to if you think I should.  I just ran out of time last night.

Here are two videos showing as clearly as I can the temperature fluctuations with the heated bed on and the lack of temperature fluctuations right after turning the heated bed off.  There is only about a second or two between the end of the first video and when the second video starts. I stopped and started recording so that I would have two separate files:
https://www.youtube.com/watch?v=DnFcbnXpij0   (Preheating takes about 7 minutes in this one.  The first temperature dip is a little bit after 8 minutes in.  The second temperature dip is at about 8:24 into the video.  The third is another 2 minutes and change after that.   So a period of a little over 2 minutes between dips)

https://www.youtube.com/watch?v=EZt28lBNxpI
I stopped recording the first video and started this one and turned off the bed.  You can see that for 6 minutes straight the nozzle temperature is stable.   I saw a 1 degree variation on the touch screen at one point.

**Comment by quarky42** [comment]
I didn't see a 220 / 110 V switch on the power supply.  Is there one that I missed somewhere?   If so, can someone direct me to it so I can make sure that the power supply isn't on 220V by mistake?

**Comment by juliandroid** [comment]
If I remember correctly, somewhere on the Creality site was written that the PSU is automatically selecting the right voltage. (Edit: I confirm, on their website it is written on the picture banner, not as a text)

Did you try setting the hotend to something like 240C to see whether time interval between heating and first drop is consistent with lower (205C) or it shorten and more often? I assume that if you have issue with the PSU it will take less time and happen more often if you draw consistently more power on that rail.

Regarding the PSU - I would inspect visually for inflated caps (turned off and not touching anything - for long time there might be 110/220V charges): 
![image](https://user-images.githubusercontent.com/7766093/58588777-1bbc1a00-8269-11e9-86ef-31fc086143e2.png)

Edit: And also check the power transistors located on the heatsink for any sign of overheating.

**Comment by juliandroid** [comment]
![20190529_233828](https://user-images.githubusercontent.com/7766093/58589705-29729f00-826b-11e9-97c5-ba5fad4fb4e2.jpg)

**Comment by quarky42** [comment]
How obvious would the signs of heating be on the power transistors?   carbon?  severe discoloration?   I didn't notice anything, but I'll look again.

I'll check the caps.  (Yes, I'll make sure not to touch the metallic bits where the angry pixies can leap out at ya!  Still a good warning, though.  ;)    I didn't notice any bulging but I can look for it specifically to be sure.

I have PLA loaded, so I am not going above about 220 degrees on the hot end.  220 just long enough doing the PID tuning causes some of the filament to crystallize.  I suspect the same effect you are talking about would happen if I drove the heated bed temperature up to 80 degrees and/or had something blowing on the bed to cool it down.   It's tied to the bed heater cycling.   My next plan was to watch the hot end voltages when the bed is heating to see what I'm getting when the power supply is under load.

**Comment by juliandroid** [comment]
I see at least 2 resettable thermal fuses, but I have no idea why if those kick-in your entire printer is not shutting down. From the datasheet ( https://www.meanwell-web.com/content/files/pdfs/productPdfs/MW/RSP-500/RSP-500-spec.pdf ) it seems it has single rail (output), so everything connected to it should shutdown as well.
Both over power and over temperature protections have automatic recovery, but there is no way for the printer to keep working for 30 seconds. 

The alternative is that only one of the power mosfets is overheating (internal temperature above 150/175C), it shutdowns and leave to the PSU with less power than expected, so the hotend is not completely turned off, but just don't have enough power to keep the temperature steady.

You said you have loaded PLA filament. What about reducing the printing speed significantly, so it becomes quite easy to maintain the hotend temperatures even when the current is reduced due to failing mosfet? I guess you'll see less temp drop than before until the mosfet recovers.

**Comment by quarky42** [comment]
Yes, printing at slower speed reduces but does not eliminate the temperature drop. One of the two prints I've had complete were at 50 percent speed and the temperature drop was still severe but not as bad... Instead of 30 degrees, it dropped 15.  With PID tuning I might be able to get that to 8 degrees... Still not acceptable but it fits in with what you are saying about the mosfets.

With that much information about what might be causing this, I can watch the power supply in FLIR and specifically look okay the mosfets while it's printing and see them over heat in real time.  Your analysis of this situation is ginormously appreciated!

Also, I have a killawatt meter.  I can watch the power going into the printer with the heated bed, the hot end, and then both the heated bed and hot end.  I also looked up the PSU data sheet and with that efficiency rating I can tell if the power supply is drawing more or less power than it should be pulling.  (Which would help rule out or indicate a short or a component pulling way more power than it should.)

We've pretty strongly narrowed this down to hardware and the fact that you're still willing to share your expertise and effort on this is really appreciated.  I may just end up buying a replacement power supply while I wait for Creality to dick around.  Does Tiny Machines sell this particular model?  (I'll check later when I'm back to a computer.)

Maybe I should open the power supply and just sit the whole thing on top of a box fan.  ;). Lol

**Comment by juliandroid** [comment]
Btw, I don't recall what was the default fan direction, but from my picture it seems to be an exhaust fan, so you had running your PSU fan in wrong direction and might caused failure. I'm using Noctua fan and I'm not even sure it has enough static pressure and flow compared to the stock, so that could be a potential problem for me (and others performed the mod too), but at least I had it running in the right direction.

I Amazon I saw a lot of "MEAN WELL RSP-500-24", so if you don't find that at Tiny Machines.

Please keep us posted with your findings here.

**Comment by InsanityAutomation** [comment]
Forget the box fan, run it submerged in liquid nitrogen! Maybe resting on some dry ice! Meanwell is usually a decent brand so it shouldnt be an issue sticking with them.

**Comment by quarky42** [comment]
@insanityautomation this guy gets it!  Go big or go home!  Lol.

I used my wattage meter last night.  The system pulls between 10 and 14 watts idle.  During bed heating only it pulls between 320 and 400 watts.  (Since resistance goes up with heat and it takes more power to heat up to higher temperatures this makes sense to me.).  During nozzle heating only it pulls about 50 watts (which makes sense for a 40 watt nominal heater cartridge).

Here is where it got interesting: With both the bed and nozzle heating it still only could pull 400 watts.  After running for a little while, it could only pull 360 watts max (thermal limiting).   So when the bed was heating, it took the whole 360 watts.  When the bed stopped heating the nozzle could catch up and it was pulling 50 watts during that time period.

So yeah, pretty sure a bad power supply.  

@juliandroid:
Having the fan blow in would not cause a failure in a good power supply.  Air flow is air flow and this power supply has thermal protection built in.  So it would protect itself.  It might not run optimally with a wrong fan direction, but this one doesn't run any differently with the fan pointing the other way.  I also read the same data sheet that @InsanityAutomation read.  It shows the fan direction on there.  It is definitely in the exhaust direction.

One concern I have: The heated bed is taking almost twice the wattage it is rated for.  The bed is a 12v 18A (220 watts) rated.  With just the bed heating up only, the system is pulling up to 400 watts.  If I replace the power supply, how much will the bed pull, then?  Will it burn out?  Is it pulling too much?

Maybe the bed has a failure that has shortened the power supply's life.

**Comment by InsanityAutomation** [comment]
That power draw sounds about right, but remember it is 24v not 12v, if you measure from either load wire on the ssr to the power supply - you should have 24v. Can measure resistance of the bed itself and I can compare it to mine tonight.

**Comment by juliandroid** [comment]
Yes, the SSR switches 24V, but after initial load it just maintain the set temperature.
One thing that is strange - you said, you have power meter and the power rating for input is 100-240V AC 5.9A. So, for my power country that is 230V*5.9A ~ 1360W and the DC output is only  24V*21A (~500W). Not sure where that difference is coming, but shouldn't be from efficiency. Anyway, the main point here is that on the outlet side I would expect to see significantly more than those 400W on the AC side, reported by  you when you heat both.
I'll try myself and see what wattage is reported :)

**Comment by quarky42** [comment]
At first all I came across was 12v heated beds for the CR-10S.  Now I am starting to see some 24V ones.

This one shows 24v and 220 watts:
https://m.banggood.com/Creality-3D-24V-220W-235235mm-Aluminum-Heated-Bed-Hot-Bed-Kit-For-Ender-3-3D-Printer-p-1335217.html

So I was mistaken about the voltage. (I wondered why they would have a 24v PSU and a 12v bed, but that's all I found at first).  But I still think the wattage is way too high on the heated bed:
https://m.banggood.com/Creality-3D-24V-220W-235235mm-Aluminum-Heated-Bed-Hot-Bed-Kit-For-Ender-3-3D-Printer-p-1335217.html

Is there any way to limit the heated bed to 220 watts from the controller side?   I doubt it.  I figure the bed just gets 24V applied to it and that's it.  However if that's the case then why does the heated bed pull more power the hotter it gets?  (If resistance is going up as it heats up and it is getting the same 24V, then the current and wattage should be going down?  I'm setting guessing my thoughts on that from before.)

**Comment by juliandroid** [comment]
I've just checked behaviour of my PSU.
I've setup the hotend to 205C and bed to 60C.
The power consumption (on the AC side) maxout at 505W. It then drop slightly to around 500W and when the temperature became closer and closer to the target it gradually dropped to around 450W (heating according to the PID).
So, most of the time it was in the range of 470-500W.

Edit: No limitation from the controller side, just the SSR and how you are going to cooling it down. Some recommends mounting the SSR on the aluminium frame few centimetres on the side, because even currently it heats up to around 70C, while mounting it on the frame drops to 45-50C. So, I would definitely move the SSR to protect it.

What I'm going to do some day is to mount 220V bed heater (needs proper grounding on the bed itself), so you are offloading the PSU dramatically.

**Comment by quarky42** [comment]
Regarding expecting to see more power being drawn: exactly!

New CR-10S Pro's come with a big aluminum heat sink on the DC Relay now.  So, mine already has that to help it shed heat.

Back to the wattage:. Exactly!  I would expect it to be able to pull up to 500 watts for a good power supply.  This one is supposed to be 89 percent efficient.  So that's a little under 450 watts usable power.  Figure about 15 watts overhead for the controller, fans, etc (based on what I saw). That is about 435 watts for all things (motors, bed heater, nozzle heater).  

Since I'm getting up to 395 watts maximum, draw, that is 365 watts usable (after conversion loss).  I've seen the bed take a lot more power than I think it is rated for.  The bed is starving the nozzle heater.  The videos show it pretty clearly, too.

Definitely not getting the full power out of the power supply.  When the bed heater is on by itself the printer pulls between 360 and 395 watts.  When both the bed heater and the nozzle heater are on, it still only pulls 360 watts (likely because the power supply has heated up and has even lower capacity at that point).  So seeing the hot end temperature drop while the bed heater is active is making more sense.

(Fully agree with the earlier estimates that the power supply is faulty... But I wanted to try and back it up with measurements based on the tools I have available.)

I'm thinking if a Mean Well 600W power supply will fit in there, I'll put that in.  They are 20mm taller.  I'd rather not run a power supply at full load.

**Comment by juliandroid** [comment]
Sounds reasonable :) Can you please upload an image of the SSR and the aluminum heat sink? I'm interested whether I can further improve my cooling of the SSR due to the planned switch to 220V AC heatbed :)

**Comment by quarky42** [comment]
Sure, I'll post a picture back when I get a chance.  In the meantime I managed to get the video clips uploaded and put into a playlist showing the power draw of my printer at different points:  
1> Initial Bed Heatup
2> A little while later, the bed is at 36C and climbing,  
3> Bed is at 54 degrees and climbing to 60C (So it is near the end of the heatup cycle)
4> Bed has reached 60C, is at steady state, so then I start heating up the hot end to 205C and we watch that heat up all the way as well as see what happens when the bed cycles.

I really like this little wattage meter.  Probably one of the best "a cheap tool to have on hand in case I need it" purchases I've made.

Here's the playlist:
https://www.youtube.com/playlist?list=PLBpIws7wZqQGts1Dw50CgzbvVv3JygCIn

**Comment by quarky42** [comment]
@juliandroid Here is that picture of the SSR with the aluminum heatsink.

I really wish Creality had rotated the heatsinks 90 degrees... both that silver one and the black ones.  If they were rotated, then the fans would cool the heatsinks down even better.

![20190530_143559](https://user-images.githubusercontent.com/48300108/58663053-d0495080-82e8-11e9-9b36-096b6d7b4eb6.jpg)

**Comment by quarky42** [comment]
@InsanityAutomation It sounds like the power draw isn't correct.  @juliandroid saw 500 to 505 watts being drawn on their system.

I do stand corrected on the voltage of the heated bed, however the 24V heated beds I find online all say 220 Watts.  That heated bed is drawing 315 or more watts by itself.   In @juliandroid's example of 505 watts that would be:

505 Watts * 0.89 efficiency = 449.5 Usable watts
449.5 - 15 watts (LCD, LEDs, Fans, Control Board) = 434.5 watts
434.45 - 40 watts heater cartridge = 394.5 watts.

So based on some number crunching I would say that the heated bed needs somewhere between 385 and 395 watts by itself.   (The heater cartridge is rated for 40 watts and I saw mine use about 40 watts pretty consistently, but it could have been 45 watts or maybe even 50.  The bed, by far, uses a lot more power.)

According to the specs of the power supply it should be able to run up to 130% of its capacity, if I'm reading it correctly.    Still, running at 500 watts on a 500 watt power supply does seem a little crazy to me.

**Comment by quarky42** [comment]
Looks like there is around  54mm height available in that base or less.   So being able to run a 600 or 650 watt power supply in there is not possible.  Damn.   Another 10mm or 20mm would be ideal.   If Creality made the base 15mm taller, then we'd have the option of switching power supplies easily.

Hmm...make a 10mm aluminum frame shim all around the base for a taller power supply?  Hmm...I could do that with extrusions.   

I guess if I did the AC bed upgrade using a DC/AC Solid State Relay, that would free up a ton of overhead. I could probably run everything else on 100 watts pretty easily.  :/   Hmm.

**Comment by juliandroid** [comment]
@quarky42 Thanks for the picture! I also see grey fan mount blowing the relay - is that printed by you or also stock now?

Yep, the PSU is rated for 504W, 24V DC max, so with 89% efficiency that makes everything except for the motors that already consumes 89% of the maximum. With all of the motors an extra 25W and you are looking at 95% load :)
From the datasheet, the AC input is 2.65A, 230V ~ 609W which related to the rated 504W make rather smaller number, so I guess there is something like extra 5% reserve, but in general it is OK for a reputable PSU to be used on 100% - after all it should work in the rated temperatures/environment and load.
Personally I prefer having something like 50% backup... and I'll achieve that by replacing the heating bed with 230V version :)

**Comment by chunter1** [comment]
> @juliandroid Here is that picture of the SSR with the aluminum heatsink.
> 
> I really wish Creality had rotated the heatsinks 90 degrees... both that silver one and the black ones. If they were rotated, then the fans would cool the heatsinks down even better.
> 
> ![20190530_143559](https://user-images.githubusercontent.com/48300108/58663053-d0495080-82e8-11e9-9b36-096b6d7b4eb6.jpg)

Thanks for the funny image of the heatsink prooving that nobody at Creality has the slightest clue of thermal management. :)
My printer came with the SSR mounted in the air, not touching anything.
I am using a thin layer of thermal conductive foil between the SSR and the printer case now. ;)

**Comment by quarky42** [comment]
A big chunk of aluminum is better than stand offs which is what they were using.  

One YouTuber noticed the relay in air and mounted it to the aluminum extrusion rail using a couple T nuts for it.  Even that was enough to cool the SSR down a great deal.

So in this case I think thermal mass is enough, but since they could get heat sinks cut and which size and direction, they should have had the fins turned 90 degrees so that the air flow would be better.

The double fan bracket seen there is my design inspired by a single fan bracket I saw online.

**Comment by juliandroid** [comment]
@quarky42  You can contact Keenovo, they are offering 310x320mm silicone pad for CR-10s Pro (110/220V versions). I don't see currently that offer in Ali, but they offered the same size previously for the price of the "normal" CR-10S :) And yes you need DC/AC SSR. The one in the printer is not appropriate :)

So, originally they didn't provide a fan for the SSR? :)

**Comment by quarky42** [comment]
@juliandroid yeah I agree about running power supplies at around 50 percent load.  They run a lot cooler yet still within their efficiency range at around there.  For me it is more like 50 to 60 percent ideal...70 percent is okay.  Anything over 80 percent is just asking for the power supply to die.  I haven't found a power supply yet that I felt was good for 100 percent load 100 percent of the time.  :)

Looking at the heated bed upgrade.  That's about $70 plus relay and insulation mat.   That's not too bad.  I want to get this power supply replaced and maybe then I'll do that next once I make sure what I bought initially works. Once it's fixed, fix it some more until it's broke... Or something like that.

**Comment by quarky42** [comment]
@juliandroid correct.  The SSR in an older version used to be installed on posts, no fan no heatsink.  

Recently they added a heatsink turned the wrong direction still no fan.  This is the version I got.   I added the second fan.

I see the heated beds on Amazon.  Not willing to go that far until the power supply situation is settled, but will keep it in mind.

**Comment by juliandroid** [comment]
BTW, adding just a small fan in front of the heat sink will help a lot, since now the fins are pretty much blocked to dissipate heat :D 

Yes, that is discouraging - all the stupid mistakes made by Creality are countless now :)
One comforting news is that the peak power consumption is rather temporary, not constant. Also this can be mitigated by Creality by heating the bed first, then the hotend. Also they can disable LCD for saving extra 1W :wink: and even put the MCU in sleep mode :rofl:

**Comment by chunter1** [comment]
Maybe of interest...
Starting at the yellow line, the diagram below shows the power-consumption of my CR-10S PRO printing a calibration cube at 40 mm/s.
Extruder is at 200°C, bed at 50°C and turned off at layer 10.

![calCube_print_power](https://user-images.githubusercontent.com/16907942/58666852-c2b9b800-8334-11e9-8e63-db296ad2efd2.jpg)

**Comment by juliandroid** [comment]
That is due to the Eco mode in the printer's menu?

**Comment by chunter1** [comment]
> That is due to the Eco mode in the printer's menu?

No, i am using Simplify3D and control everything from there.
By the way... If you don't have it already... i can highly recommend using a Raspberry Pi with Octoprint.
It's really increadible convenient :)
Gonna integrate the Pi into the Creality case next.

**Comment by juliandroid** [comment]
Yep, I'm having an older Pi that I want to use it for OctoPrint precisely :) So many things I want to do, so less time to do them :)

**Comment by quarky42** [comment]
The inside of my power supply and the DC relay when the heated bed has reached 60C:

![Mean Well Power Supply Temperatures at 60C Heated Bed flir_20190530T155037](https://user-images.githubusercontent.com/48300108/58669709-24a8fc00-82fa-11e9-8332-937768c3e8fe.jpg)

Edit:  Here's the temperature gradient turned off:
![Mean Well Power Supply Temperatures at 60C Heated Bed 20190530T155037](https://user-images.githubusercontent.com/48300108/58670029-ef50de00-82fa-11e9-9e8e-75bb1a3d384d.jpg)

**Comment by quarky42** [comment]
That inductor coil though... 70.8 degrees C.  Yeah, it's getting a little too hot in there at 360 to 395 watts of power draw.

**Comment by juliandroid** [comment]
From temperature side, I don't see nothing imminent that could cause the problem. Maybe it is time for PSU replacement. I cannot imagine that the problem is coming from the heatbed (both to consume less than rated and also stop working periodically), although for example I've seen sensors working perfectly while cold (normal resistance) that goes to "infinite" when heated and that cycles over and over again :)

**Comment by quarky42** [comment]
Yeah I've reached out to the seller, Creality, and Mean Well with the information.  The PSU has a 3 year warranty and Mean Well has a USA based website at least so maybe they can do a warranty exchange in a reasonable amount of time.

I did notice some clicking (rapid cycling on/off) of the solid state relay though.   I think it had more to do with the PSU reaching it's faulty overload limit.  The SSR relay led and the power led on the PSU both blink rapidly for a few seconds.  I'll see about getting that video loaded.

At this point I've just been adding info to the thread because I'm able to do a bit more testing and in the hopes that once I get a replacement PSU that the information here ends up helping someone else that comes across it.

I appreciate the thoughtful discussion that everyone has contributed to this far!

**Comment by InsanityAutomation** [comment]
Rapid blinking is expected in pwm control as it gets close to temp. PID takes over within 10c of setpoint.

**Comment by quarky42** [comment]
Solid State Relay and PSU LEDs rapidly blinking when the bed gets up to temperature:
https://youtu.be/cQzmc1puYko

From the video it looks like just before the relay turns off, it starts to cycle rapidly.  Does this hint at a control board problem?

I don't believe I've seen this when the relay is turning on... Just when it is turning off, so far.

I'll see if this behavior goes away after the psu is replaced.

**Comment by quarky42** [comment]
@insanityautomation it is happening only in the last second or two of bed heating.  Not the last 10C.

Also it is causing the power supply LEDs to cycle in time with the SSR.  So is this another indication of a weak PSU?   

Do your PSU LEDs blink just before or as the set point is reached?

**Comment by quarky42** [comment]
Creality has been most unhelpful and replies only to messages they feel like it.

I provided what I feel is adequate proof that there is a problem with the power supply.  But that wasn't good enough for them.  They want to see the voltages drop.  So while I'm getting setup to measure the hot end voltage I notice that the power supply dc wires are mirror images of each other.

Would someone be willing to share a picture of their working cr-10s pro power supply connections?

What I'm seeing is that the 24v exhaust fan is on pin 1 and 6.  The bed is on pin 2 and 5.  I assume the hot end is on 3 and 4.    Either it doesn't matter what pins the go on as long as the positive and negative polarities are respected or maybe there is an issue with having the ground return to the wrong pin relative to the positive.  (Not having wired any power supplies before I would assume that pin 1 goes with pin 4, 2 with 5 and 3 with 6, but maybe that isn't necessary.)

Here is what my power supply connections look like:
![20190609_145432](https://user-images.githubusercontent.com/48300108/59164728-7c035500-8ace-11e9-8995-38eb55c72659.jpg)

**Comment by juliandroid** [comment]
Looks like the same to me. Do you have any improvment? New PSU?

![psu](https://user-images.githubusercontent.com/7766093/59439882-b3d80880-8dfe-11e9-81c4-fa10f4cb84cf.jpg)

**Comment by chunter1** [comment]
>  Either it doesn't matter what pins the go on as long as the positive and negative polarities are respected or maybe there is an issue with having the ground return to the wrong pin relative to the positive. (Not having wired any power supplies before I would assume that pin 1 goes with pin 4, 2 with 5 and 3 with 6, but maybe that isn't necessary.)

The order of the connections does not matter at all.
Just check if the supply provides constant 24V in every printing stage.

**Comment by quarky42** [comment]
Thank you VERY much for the sanity check.   Glad to see it is the same.   Finally the seller AND creality agreed to send a new power supply.  It only took roughly a month to convince them from when I was first convinced (with the help of this thread and everyone contributing to it) that there was a power supply problem.

My "sunshine parting the clouds of confusion moment" was likely when someone confirmed that their unit pulled 500 - 505 Watts with the same settings while mine pulled a max of 390 watts with a cold start and later 360 watts when it had been running for a while....   That right there was a dead give away that the power supply couldn't handle the load (when added to the fact that the hot end by itself pulling 40 watts plus the ~10 watts of overhead worked fine without the heated bed running).   

Later on I even measured the PSU voltage while the heated bed was on:   9 - 13V while the heated bed was active....  yeah that PSU is failing under load.   Still even after all this information they still wanted more proof.   Ugh.   Now to wait the 3 weeks for them to ship me a new one.

I went ahead and ordered a 750W silicone 310mm AC heater and a brand name SCHNEIDER ELECTRIC DC controlled AC relay (40A max), and bed insulation.   I'll may have that conversion done this weekend (and if not then next weekend for sure) and the new power supply will just be a backup for when this one finally eats itself.   Maybe my weak PSU will last long enough at 50 watts output to actually get some prints while I wait for the slow boat from China to arrive.    

My only other challenge is that I would like to somehow AC current limit the bed to 6 amps maximum.   If I do figure out a good way to do that, then I should be able to drive that bed up to a maximum of 200 degrees (really I'm looking for between 130 and 150 degrees, but if I can make it safe for the heater's max temperature of 200 degrees, that provides the kind of margin of safety I like to design for.)   I'm going to use double insulation on the bottom of the bed AND a layer of insulation on top of the printer housing below the bed to keep the heat directed upward.   

I'm intending to eventually design an enclosure that sits over the top of the printer as well.  The design of which would leave the base outside / below the enclosure so that it would still draw in cool air and keep the internals cool.   If/when I do this, I'll remote the extruder motor out using the Zesty Nimble to get the extruder motor out of the chamber.  I might have to put fans on the Z and X axis motor to keep them from heating up too much and/or drive them slower when printing at higher temperatures.    I intend to print in PETG mostly, so that won't be a problem, but I would like the ability to print in POM and/or experiment with high temperature materials.  I already have the all metal hot end for it.

I am really thankful for everyone's contributions to this thread.  I can't imagine how long this would have taken to get Creality to send a new PSU without this help!

**Comment by juliandroid** [comment]
If the goal is safety you can go another way:
For example there are temperature limiting switches/fuses: https://www.aliexpress.com/item/KSD9700-250V-5A-15-155-Degree-Bimetal-Disc-Temperature-Switch-Normal-Close-Closed-Thermostat-Thermal-Protector/32860802842.html?spm=a2g0s.9042311.0.0.5b7b4c4doCsbeN

This is for 230V@5A max, which theoretically is way above those 750W. If you pick 155C Normally Closed version then, going above 155C will cut the power. Obviously you have to place it between between the bed and the pad.

However I'm not sure whether that will be enough, due to the hysteresis, i.e. 155C might cut too early and not be able to actually go above, for example 110-120C. So, I would go with higher "peak" temperature.

**Comment by quarky42** [comment]
@chunter1  yep, our replies crossed paths   (more like I was still writing mine while yours came in).   I'm caught up with that understanding.   It had to be either / or.  ;)   I'm just glad to have a new power supply coming my direction finally, and in the mean time maybe an AC bed will work better anyway.  (Even if I was going to the AC bed, I would want a good / proper power supply because I might use it for something else.)

**Comment by quarky42** [comment]
> 
> 
> If the goal is safety you can go another way:
> For example there are temperature limiting switches/fuses: https://www.aliexpress.com/item/KSD9700-250V-5A-15-155-Degree-Bimetal-Disc-Temperature-Switch-Normal-Close-Closed-Thermostat-Thermal-Protector/32860802842.html?spm=a2g0s.9042311.0.0.5b7b4c4doCsbeN
> 
> This is for 230V@5A max, which theoretically is way above those 750W. If you pick 155C Normally Closed version then, going above 155C will cut the power. Obviously you have to place it between between the bed and the pad.
> 

I follow the logic there.  The one problem with that is if the limiting switch / thermal fuse loses contact with the bed, but I could see using both a current limiting device and a thermal switch.   Nothing wrong with safety that would require multiple failures before catastrophic failure occurs, especially considering these kinds of components aren't that expensive.

**Comment by quarky42** [comment]
Ok, maybe I'm a geek, nerd or whatever, but I thought this was really helpful / neat to read in relation to my current limiting concerns:  
http://www.resistorguide.com/ptc-thermistor/

So maybe a 7A equilibrium PTC thermister (or maybe a little under 7A like 6.6A because I don't think I'll ever need the full 200 degrees of the pad... maybe around 180 degrees max).

**Comment by juliandroid** [comment]
I don't thing those PTC thermistors are intended for high current, high voltage. What you really need is just a normal (ceramic) fuse and fuse holder. Let's say your pad is consuming 6.2A, then I would go with 8A slow fuse and if short hapens this is the first thing to blow. However, as I menitioned before you have to mandatory ground the heat bed directly to the ground lead coming from the power cable. This is more important, due to the nature of the pad - it is just a large resistor/wire that if it breaks it will just have very large resistance, but if that touches the metal plate it might electrocute you if there is not a proper grounding.
An extra life saviour could be a WiFi temperature/fire alarm that will turn on and also send you notification.

**Comment by quarky42** [comment]
>  
> I don't thing those PTC thermistors are intended for high current, high voltage. 

Not true.  There are PTC Thermistors that are rated for 250VAC and 6A or 7A.  I am referring to specific Thermistors that I've gone and looked at.   There are all kinds of Thermistors out there.  The one danger is that the "best" have a tolerance of +/-15% and the silicone heater pad says not to exceed 200C (so I'm assuming that is 750 watts for 200C) and certainly it would be bad to exceed 6A anyway, so preventing that is a good idea. There was an error in my math earlier, so that would actually be 6A max (not the 7A I was talking about), but with a +/- tolerance then I might need a thermistor that is rated for 6A/1.15 or about 5.2 Amps or 5Amps.   That would in turn limit my silicone heater (at worst case scenario) to about 650 Watts.... which doesn't sound bad at all, to me.   That should still be quite hot.

Finding fuses for 7A or 8A can be expensive.  These PTC Thermistors are cheap and I could buy enough of them for $10 that I could just test them all and use the "best behaved one".

This should be combined with a bimetal disconnect to prevent going over 200 degrees C as well.  These two safety things would work in conjunction really well.  The idea of using a bimetal temperature disconnect was a good one, but I wouldn't want to use that by itself.  Maybe even use 2 of them in series so that if one fails / falls off, the other would still work.

> An extra life saviour could be a WiFi temperature/fire alarm that will turn on and also send you notification.

If I am away from home, this would do me no good.  If I am asleep, this would do me no good unless I make the notification really loud.    However, if I did something like this, it would be to trigger a complete power shutdown of the printer at the first sign of smoke.  The sensitivity of this could be a problem or it might be able to be dialed in depending on the sensors available to do this job.   Further more, it could be used to trigger or deploy some sort of CO2 based suppressant, especially if this is inside of an enclosure.

**Comment by juliandroid** [comment]
@quarky42 That is interesting. I was looking at typical EU suppliers for high power PTC thermistors but I couldn't find one, same issue with Aliexpress. Can you post a link, so I can take a look at their datasheet?

For the simplest solution with ceramic fuse and holder, you can buy 5pcs for $1-$1.5 total, the holders are also the same price. Resettable fuses for those cases makes no sense - I would rather have fuse blown and then to manually inspect the issue than having something that might manifest issue, but then hide the problem behind the resettable fuse.

Also you have thermal runaway protection for both hotend and heatbed, so those will kick in if your temp sensor is out of order (too low or too high resistance).

Regarding the WiFi alarm, the idea is that is has strong siren for local notification and also an extra interface for performing tasks, like send IFTTT command or to stop your smart outlet. I have smart home automation, so I have all of those features, like Fire alarm that is connected to my hub, also temperature sensors and smart power plugs. At some point of time, I'm going to connect those to the printer :)

**Comment by quarky42** [comment]
Mouser seems to have one of the best "searches" that let's me try and figure out what range of PTC Thermistors would apply (in this case called Inrush Current Limiters.)   I have found several places yesterday, but I closed those tabs and don't feel like digging through my browsing history.   I'll just look up some more when I get a chance and update this post.

To start:
Mouser Search:  Inrush Current Limiters https://bit.ly/2MNQzFF
Selected 5.1A through 6A.  Because the tolerance is between 10 and 20%, that means with a 6A absolute maximum, either buy enough 6A or just under 6A until you find one that is the spec you want, or just use one that is 5.1 or 5.2A with a 15% possible max.   Maybe by choosing a handful of the the 5.5A ones and maybe a handful of the 6A ones, you'd find one that was for sure less than 6A.   Would have to test them, though.
Edit:  Maybe a 6A Circuit Breaker / Manual Heated Bed Switch:
https://www.digikey.com/product-detail/en/te-connectivity-potter-brumfield-relays/W51-A121B1-6/PB2797-ND/2324560
So if it gets to 6A for too long the breaker just trips and only the bed turns off.

Re: Ceramic fuses.  There is room for a max limit and a blown fuse to replace, but I'd have more time to see what is going on if I saw the temperature hitting a lower limit and not going any higher.  I could measure the current.  I could measure the voltage.... if a fuse blows, all I know is that fuse rating was exceeded.  I didn't see 6A fuses at first, but I do see some AGC fuses like that.  I'm not sure what their tolerances are though.    Perhaps there is room for both...assuming that the AGC is precisely 6A or just a slight hair under, but before the fuse I would want the PTC Thermistor / Inrush Current Limiter.

Software thermal runaway protection is great.  That's just another layer of protection.  I like the idea of having multiple layers of protection when the risk is fire.

I think we see eye to eye on the WiFi alarm now.  I was thinking along the lines of having control some smart devices like a smart outlet and/or a fire suppression system.  It could be done in stages:
A> Temperature X || Smoke, turn the power off to the printer, Send alerts that something is wrong.  (If this was eventually used in conjunction with print resume, this wouldn't necessarily cause a print failure, it would just tell you that something was sketchy and it was better safe than sorry.
B> Temperature Y && Smoke, turn off the power to the printer and trigger the fire suppression actuator, and alert.
C> Temperature Z, send an alert to kiss it all good bye.
X < Y << Z in this case.  ;)

As a bonus, if you have the right kind of smart power plug connected, you not only get on/off control and information, but you also can see how much power you're consuming, so if you are making prints that people are buying you can know what your energy costs are pretty easily.

**Comment by juliandroid** [comment]
Btw, on the first Mouser serch, those are Inrush Current Limiters, so they have high resistance (5-10ohms) to allow for graceful/soft start. That can help increasing life of the heatbed. However, once heated up (in seconds), then the resistance goes down to significantly lower values (sub 1 ohm). If you go above their thermal limits (~170C), you are going to damage the element, so effectively you are going to use them as fuses, but with significantly poor and unsafe "mode" - after all they are not intended to replace a fuse.

**Comment by quarky42** [comment]
Then I had it backwards, because what I understood was that the resistance went up as the heat did. I understand what you explained once I double checked the data sheet.   What are oil filled heaters using then?  They switch at temperature, at high current and don't catch fire... Just those bimetal switches?  (I'm still concerned about them coming detached from the bed and want to limit the current directly to 6a... So maybe that breaker then.)

**Comment by juliandroid** [comment]
The bimetal switch is going between the heater and the bed, so worst case - it will make contact with the pad only, which is the heating element (i.e. maximum temperature). However I don't know whether you can find such that can work at higher temperature (above 155C). Another issue is the glue/stick material on the pad side that should be able to withstand those temperatures.

**Comment by quarky42** [comment]
The manufacturer claims 200 C is the max it can handle.  

There are bimetal switches for 200C.  Their accuracy is questionable.  So maybe 190 would be better.  I didn't see how they could be held into place because I only knew of the thicker round style. I do see some smaller flat ones, but they only go up to 5A.  So you would need to interrupt the relay DC control like instead of cutting the AC power of the bed heater directly.

**Comment by juliandroid** [comment]
Btw, what silicone heatbed you bought as replacement for CR-10s Pro? Did you find one 310x320?

**Comment by quarky42** [comment]
Nope.  The  310mm CR-10 / CR-10S one I bought ended up being for a different mounting pattern, so I'm sending it back.   I have contacted Keenovo asking them to make one.  Their response is "we already sell on for the CR-10S Pro" and they pointed me to the one that has the same specs as the one that failed to fit.

I have provided and one other silicone heater manufacturer detailed measurements and a suggested spec for one that properly and safely fits the CR-10S Pro Heated Bed Model CR-X.  The detailed dimensions of the heated bed with cut outs for the bed clamps:
![Creality CR-10S Pro Heated Bed Model CR-X Detailed Dimensions](https://user-images.githubusercontent.com/48300108/59567270-c2096d00-9028-11e9-9e9f-99a1ef2c559c.png)
I made the screw holes a little over 2x the actual diameter in the above drawing.  The actual diameter 
of the screw came out to be 3.7mm, the hole about 4mm just by rounding and then I doubled that for the drawing above.

AND:   A Simple Bed Heater that would fit between those clamps 310mm wide in the X axis by 305mm tall in the Y axis (307.3mm max, but there really should be some clearance, so nominally 305 or 306mm)
![Creality CR-10S Pro Simple Bed Heater Dimensions](https://user-images.githubusercontent.com/48300108/59567271-c2a20380-9028-11e9-9df5-dbac72722d29.png)
In this drawing I made the screw holes 3 times larger than the actual holes because you don't want the traces of the heater anywhere near the screws for a possible short.  Somewhere between 8mm and 12mm is fine.  The closer the tolerance, the more exact the heater has to fit the bed.

I have also asked them to put the cables coming out of the rear left corner when the bed is in the installed position so that we can retain the factory strain relief.  (That would be the rear right corner when the bed is upside down.)   

The USA manufacturer (might be an importer...not sure.  They claim to use USA materials.) has shown interest in releasing one specifically for the CR-10S Pro.  I hope they will continue forward with it.  I will certainly buy it.   Keenovo so far has only sent the one response indicating that the one they sell is for the CR-10S Pro, which I highly doubt.  Does anyone here have the Keenovo and can you confirm how well it fits the hole mounting pattern on the bottom of the printer?

**Comment by juliandroid** [comment]
The distance between the holes on the X-axis was around 250mm, while on the Y was around 240mm.
So in theory if they offer such distances, even if the pad is 10mm shorter (5mm on both sides of the X) it should be ok.

I have one 240x240 distance and I have calculated that I need to extend the diameter of the holes with 2-3mm in order to be able to mount it and depend how much spacing is available between the heater inside the the silicone border around. Worst case I'll damage the pad and I'll have to buy new one appropriate for CR-10s Pro.

Do you have links to the model they claim is compatible with CR-10s Pro and how long do you think you have to wait to get the new pad?

Please keep me in touch with thr progress.

**Comment by quarky42** [comment]
The one I tried doesn't have a compatible one that I could find.

The one I'm working on with another company won't be ready to for a week roughly and it is a prototype.

Keenovo replied back and said this one will fit the Pro:
https://keenovo.store/products/keenovo-silicone-heater-pad-for-creality-cr10-3d-printer-build-plate-heatbed-heating-upgrade?variant=13977831014455

They said:
"Yes, it’s wider than it’s deep. 
We enlarged the holes on the x directions so it fits the CR10 as well as the CR10S Pro. 
Please do choose the version specified for CR10S Pro. 
   
The picture is only for reference only."

So it should fit.

When I checked their website, I didn't see that one listed there the first time.  I might have missed it or maybe they added a specific listing for it.  Either way I ordered one.  I need my printer working asap.

**Comment by juliandroid** [comment]
Yes, that is correct, indeed.
I've asked Keenovo before and they said that it is possible to do it for CR-10s Pro. Later a guy in facebook group asked about for this and I referenced him to the Keenovo representative and later he said, Keenovo asked him for correct dimensions, so they could produce that pad for him as a test customer. The result is actually reusing the pad for CR-10s by extend the hole in oval form, so it can fit both CR-10 and Pro model, so I guess this is exactly the same product that you are going to get.

So, you are actially ordered this one from Keenovo, right?

**Comment by quarky42** [comment]
I did order that one this morning once they provided the explanation I quoted.  My concern was that the holes were too close to the screws to be safe in the other one (different brand) I received.  Since they elongated the holes in this model, I think that is a fair solution.

**Comment by quarky42** [comment]
I'll still post back if / when that other group comes up with theirs.

**Comment by quarky42** [comment]
I took all the mounting hardware off my heated bed and installed the Keenovo silicone heater.  The holes did line up.  If it were up to me, I would make the heater about 5 or 8mm shorter in Y direction so there is more room to use binder clips to hold the bed.  I can barely get the clips to hold with the current tolerances, but a little more would really help.    Looks like the other company has developed theirs based on my drawings / measurements of the bed ansting clamps for high temperature applications.  I will use this first bed for a spring steel style build plate magnetically held for normal temperature applications  (PETG and PLA).   

Anyway, I just wanted to report back that I have the printer printing.  I have to diagnose some stringing (PETG) and some first layer adhesion problems (PLA).  I have been able to get prints with both, but I likely need a higher flow rate for my PLA at this point based on what I'm seeing.   This AC bed heats up super fast.   I can get it to 60 degrees in 1 minute.  80 degrees in a minute and a half.  I really like it so far.

**Comment by quarky42** [comment]
I've had a great discussion here, tremendous help from everyone, and in the end the problem was 100% hardware.   I still haven't received my new power supply from the seller / Creality but it should be here in the next week or two, supposedly.   

I should probably have closed this issue sooner when we figured out together that the power supply was  not up to the job.  I did find the continuing discussion of other loosely related things to be incredibly helpful though.  I thank everyone that contributed to this and I thank the mod/developers for being patient and leaving this active discussion open for as long as they did.
